src/core/plugin_manager.h

```diff
@@ -82,33 +108,92 @@ class nixlPluginManager {
     nixlPluginManager(const nixlPluginManager&) = delete;
     nixlPluginManager& operator=(const nixlPluginManager&) = delete;
 
-    std::shared_ptr<const nixlPluginHandle> loadPluginFromPath(const std::string& plugin_path);
+    void
+    loadPluginsFromList(const std::string &filename);
 
-    void loadPluginsFromList(const std::string& filename);
+    // Load a specific backend plugin
+    std::shared_ptr<const nixlBackendPluginHandle>
+    loadBackendPlugin(const nixl_backend_t &plugin_name);
 
-    // Load a specific plugin
-    std::shared_ptr<const nixlPluginHandle> loadPlugin(const nixl_backend_t& plugin_name);
+    // Load a specific telemetry plugin
+    std::shared_ptr<const nixlTelemetryPluginHandle>
+    loadTelemetryPlugin(const nixl_telemetry_plugin_t &plugin_name);
```

Comment:
The method should be private, as it's only used inside another class method.
There is no need to return `shared_ptr` from the method. Maybe just true/false.